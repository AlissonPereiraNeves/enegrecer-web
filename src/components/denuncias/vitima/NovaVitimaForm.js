import React from 'react';
import CheckBox from '../../comum/checkbox';
import * as ConstantesCSS from '../ConstantesCss';
import Nome from '../../comum/nome';
import Genero from '../../comum/genero';
import Telefone from '../../comum/telefone';
import ComboboxRaca from './../../comum/comboboxRaca';
import Data from '../../comum/data';
import Endereco from '../../comum/endereco';

const NovaVitimaForm = () => (
  <div>
    <div className="row">
      <CheckBox
        state={'souAVitima'}
        divClasse={'col s12 m6 l6'}
        id={'souAVitima'}
        label={'Marque se você é a vítima'}
      />
    </div>
    <div className="row">
      <Nome id={'nome-vitima'} state={'nomeVitima'} />
      <Data
        state={'dataNascimentoVitima'}
        id={'dataNascimento'}
        label={'Data de Nascimento'}
        divClasse={`${ConstantesCSS.CLASSES_DIV_INPUT} col s12 m6 l6`}
      />
    </div>

    <Endereco estadoState={'estadoVitima'} cidadeState={'cidadeVitima'} />

    <div className="row">
      <Telefone
        divClasse={'input-field col s12 m6 l6'}
        id="telefone"
        state={'telefoneVitima'}
        label={'Telefone'}
      />
    </div>

    <div className="row">
      <Genero
        state={'generoVitima'}
        id={'vitima-genero'}
        divClasse={'col s12 m6 l6'}
      />

      <ComboboxRaca
        state={'racaVitima'}
        id={'raca'}
        divClasse={'col s12 m6 l6'}
        somenteRacasVitima
      />
    </div>
  </div>
);

export default NovaVitimaForm;
