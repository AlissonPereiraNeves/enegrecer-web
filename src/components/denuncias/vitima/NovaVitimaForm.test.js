import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import NovaVitimaForm from './NovaVitimaForm';

describe('NovaVitimaForm Component', () => {
  it('deve possuir estrutura definida ', () => {
    const detalhamentoDenuncia = shallow(<NovaVitimaForm />);
    const detalhamentoDenunciaJson = toJson(detalhamentoDenuncia);
    expect(detalhamentoDenunciaJson).toMatchSnapshot();
  });
});
