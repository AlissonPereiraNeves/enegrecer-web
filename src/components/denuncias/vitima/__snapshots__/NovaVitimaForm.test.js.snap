// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NovaVitimaForm Component deve possuir estrutura definida  1`] = `
<div>
  <h3>
    Me conte um pouco sobre a vítima
  </h3>
  <br />
  <checkBox
    id="souAVitima"
    label="Marque se você é a vítima"
    state="souAVitima"
  />
  <div
    className="row"
  >
    <nome
      id="nome-vitima"
      onChange={[Function]}
      state="nomeVitima"
    />
    <Data
      divClasse="input-field col s6"
      id="dataNascimento"
      label="Data de Nascimento"
      state="dataNascimentoVitima"
    />
  </div>
  <div
    className="row"
  >
    <telefone
      divClasse="input-field col s6"
      id="telefone"
      label="Telefone"
      state="telefoneVitima"
    />
    <combobox
      divClasse="col s6"
      id="cidade-vitima"
      itens={
        Array [
          "Belo Horizonte",
        ]
      }
      label="Cidade *"
      state="cidadeVitima"
      valorPadrao=""
    />
  </div>
  <div
    className="row"
  >
    <genero
      divClasse="col s6"
      id="vitima-genero"
      state="vitimaGenero"
    />
    <comboRaca
      divClasse="col s6"
      id="raca"
      somenteRacasVitima={true}
      state="raca"
    />
  </div>
</div>
`;
